### PR TITLE
Force write bytes to happen on correct thread

### PR DESCRIPTION
**Pull request by DonLakeFlyer**
* Fix for #9022
* The #8835 change was incorrect in that it caused the call to QIODevice::write to possibly happen on a different thread than the link itself was on. This worked for some link types and not for others.
* In order to fix this I moved back to signaling the write bytes to the link. This will cause them to be queued and show up on the correct thread if the threads differ.

### CODE REVIEW

File: src/PositionManager/PositionManager.cpp [hunk 1]
```diff
@@ -39,7 +45,34 @@ void QGCPositionManager::setToolbox(QGCToolbox *toolbox)
 
    if (!_defaultSource) {
        //-- Otherwise, create a default one
+#if 0
```
Comment (keith-bennett-airmap) [inline src/PositionManager/PositionManager.cpp:48]: `#if 0` is an ugly wart

File: src/comm/LinkInterface.cc [hunk 2]
```diff
@@ -80,10 +82,7 @@ void LinkInterface::_freeMavlinkChannel()
 
 void LinkInterface::writeBytesThreadSafe(const char *bytes, int length)
 {
-    QByteArray byteArray(bytes, length);
-    _writeBytesMutex.lock();
-    _writeBytes(byteArray);
-    _writeBytesMutex.unlock();
+    emit _invokeWriteBytes(QByteArray(bytes, length));
```
Comment (keith-bennett-airmap) [inline src/comm/LinkInterface.cc:85]: From [documentation](https://doc.qt.io/qt-5/qbytearray.html#QByteArray-1):
> QByteArray makes a deep copy of the string data.

:+1:

**Review by keith-bennett-airmap — COMMENTED** [review]
It looks like the changes to to `src/comm/LinkInterface.h/cc` are the only changes needed. The other changes look like they're unrelated

**Comment by DonLakeFlyer** [comment]
This has two commits in it. The second one is to make QT_FATAL_WARNING usable again. They were required for me to be able to debug this problem. They are not required to fix the joystick problem. They are just required to make this type of debugging possible again.

**Comment by DonLakeFlyer** [comment]
I'll squash the android change together.

**Comment by DonLakeFlyer** [comment]
The #if 0 is to leave the old code in, both to explain the Qt strangeness and also under the hope that at some point Qt fixes this and we can just move back to that simpler direct call.

**Pull request merged**